nextcloud23: 23.0.8 -> 23.0.9, nextcloud24: 24.0.4 -> 24.0.5

Ma27 opened the pull request:
###### Description of changes

https://nextcloud.com/changelog/#23-0-9
https://nextcloud.com/changelog/#24-0-5

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Ma27 committed:
nextcloud23: 23.0.8 -> 23.0.9

ChangeLog: https://nextcloud.com/changelog/#23-0-9

Ma27 committed:
nextcloud24: 24.0.4 -> 24.0.5

ChangeLog: https://nextcloud.com/changelog/#24-0-5

yayayayaka commented:
The `with-mysql-and-memcached` test failed: [nextcloud.log](https://github.com/NixOS/nixpkgs/files/9539860/nextcloud.log)

Ma27 commented:
Will look later today.

Ma27 commented:
OK, so I managed to fix the test by reverting https://github.com/nextcloud/server/pull/33513/commits locally.
Will try to find out later what's exactly wrong with the patch.

Ma27 commented:
So... since I'm procrastinating anyways, I just found the cause: Nextcloud tries to recreate a custom DB user with the supplied DB user and generates a secure DB password. Apparently, the commit I linked above always alters the DB password for the MySQL user and because we set the "old" db password in override.config.php, so the db connection fails.

I think it was sheer luck that we haven't encountered this before because the regeneration (AFAIU) would've always happened if the DB user had elevated rights. Hence I'm inclined to patch away the entire behavior since it's inherently incompatible with how we manage Nextcloud in our module.

Also, I guess I'll report this as a bug to Nextcloud because always altering the db password of a user seems wrong to me.

Ma27 committed:
nextcloud: drop password regeneration behavior

While updating to 23.0.9/24.0.5[1], it was discovered that Nextcloud
silently changes db passwords in some cases (in case of MySQL in **all**
now).

This is inherently incompatible with our module. Further context is
provided in the patch file attached to this commit.

[1] https://github.com/NixOS/nixpkgs/pull/190646, see comments below

Ma27 committed:
nixos/nextcloud: fix a deprecation warning in the tests using redis

Ma27 commented:
@yayayayaka should be fixed now :)

yayayayaka reviewed (commented):
Tests pass, thank you!

While we are at it, change the redis configuration in `nixos/tests/nextcloud/with-declarative-redis-and-secrets.nix` as well. Otherwise, you have my approval! :)

Ma27 commented:
Good point, done!

SuperSandro2000 commented on nixos/tests/nextcloud/with-declarative-redis-and-secrets.nix:
```diff
@@ -47,9 +47,7 @@ in {
-      services.redis = {
-        enable = true;
-      };
+      services.redis.servers."".enable = true;
```
Should we name this?

```suggestion
      services.redis.servers."nextcloud".enable = true;
```

Ma27 replied:
done.

Ma27 commented:
Filed https://github.com/nextcloud/server/issues/34123 for the mysql problem.
